solution/utils.go

```diff
@@ -0,0 +1,8 @@
+package solution
+
+func ReverseSlice(arr []PlanningSolution) []PlanningSolution {
```

Comment:
Maybe we can make this function generic and put it in the `utils` package? For example, `func ReverseSlice(arr []interface{}) []interface{} {...} `

Reply:
sg. I actually tried to use interface as arguments, but it somehow doesn't work as expected. I would get this error in VScode.
![Screen Shot 2022-05-02 at 4 57 04 PM](https://user-images.githubusercontent.com/54508739/166344352-de4c32f0-1207-45b8-ae6b-291e1ab0af4c.png)

Reply:
I have found there are some other examples, using the `T` type to indicate its generic data type. However it requires Go version newer than 1.18, and we are 1.16 I believe.
![Screen Shot 2022-04-30 at 5 59 14 PM](https://user-images.githubusercontent.com/54508739/166344444-587614b3-9d61-48b0-9327-ba61e79426a7.png)

Reply:
We don't need generics here. Refer to this example I wrote: https://go.dev/play/p/aaW5z6Uc8GS

Reply:
I see, but is your example for `maps` not `slice` in golang? 
I have also found an article about interface slice, but it also suggests the following input and output don't work right away.
`func ReverseSlice(arr []interface{}) []interface{} {...}`

Reply:
Makes sense, then we should just name the method ReversePlans and put it in the same file

Reply:
Sure, I just updated the function in 21c6d64